src/audio/module_adapter/module_adapter.c

```diff
@@ -184,6 +184,7 @@ static int module_adapter_sink_src_prepare(struct comp_dev *dev)
 	uint32_t num_of_sources = 0;
 	uint32_t num_of_sinks = 0;
 	int ret;
+	int i = 0;
```

Comment:
superfluous initialisation in both cases. And I think we're now building for C99 by default...

Reply:
> According to the following documentation, xt-xcc does not support C99.
> 
> > > > > https://thesofproject.github.io/latest/introduction/index.html
> > > > > Cadence xt-xcc or Cadence xt-clang?
> 
> It depends on the platform. As of June 2020, most platforms supported by SOF rely on xt-xcc. Going forward, all newer platforms require xt-clang. The gcc frontend doesn’t support unusually large registers, hence the move to xt-clang.
> 
> Note that xt-xcc does not fully support C99. xt-clang does.
> 
> Local test result: 1.Add the "-std=c99" C_FLAG , we can support C99, but more build errors will be generated. 2. use xt-clang could fix current issue, but need remove unused function.

According to the following documentation, xt-xcc does not support C99.

https://thesofproject.github.io/latest/introduction/index.html
Cadence xt-xcc or Cadence xt-clang?

It depends on the platform. As of June 2020, most platforms supported by SOF rely on xt-xcc. Going forward, all newer platforms require xt-clang. The gcc frontend doesn’t support unusually large registers, hence the move to xt-clang.

Note that xt-xcc does not fully support C99. xt-clang does.

Local test result:
1.Add the "-std=c99" C_FLAG , we can support C99, but more build errors will be generated.
2. use xt-clang could fix current issue, but need remove unused function.

Test code: [https://gist.github.com/hailongfan-mtk/c474e9b043e1a7df5f24e336369f46a5](url)

Reply:
> superfluous initialisation in both cases. And I think we're now building for C99 by default...

According to the following documentation, xt-xcc does not support C99.

https://thesofproject.github.io/latest/introduction/index.html
Cadence xt-xcc or Cadence xt-clang?

It depends on the platform. As of June 2020, most platforms supported by SOF rely on xt-xcc. Going forward, all newer platforms require xt-clang. The gcc frontend doesn’t support unusually large registers, hence the move to xt-clang.

Note that xt-xcc does not fully support C99. xt-clang does.

Local test result:
1.Add the "-std=c99" C_FLAG , we can support C99, but more build errors will be generated.
2. use xt-clang could fix current issue, but need remove unused function.

Test code: [https://gist.github.com/hailongfan-mtk/c474e9b043e1a7df5f24e336369f46a5](https://github.com/thesofproject/sof/pull/url)

Reply:
@hailongfan-mtk so Intel deprecated a lot of the older legacy platforms that used older xt-xcc in CI and this is probably why new PRs are passing with C99 changes. It sounds like we need to enforce the minimum agreed C version on the CI .